src/SFA.DAS.LearnerDataMismatches.Domain/LearnerReport.cs

```diff
@@ -8,8 +7,7 @@ public class LearnerReport
         public (string Id, string Name) Learner { get; set; }
         public (string Id, string Name) Employer { get; set; }
         public (string Id, string Name) Provider { get; set; }
-        public IEnumerable<CollectionPeriod> DataLocks { get; set; } = new List<CollectionPeriod>();
-
-        public bool HasAnyDataLocks => DataLocks.Any();
+        public bool HasDataLocks { get; set; }
+        public Dictionary<int, List<CollectionPeriod>> DataLocks { get; set; }
```

Comment:
Should the first template type be `AcademicYear` to express it's a list of year's worth of collection periods?

Reply:
Since the keys are formed as part of grouping of the data by the AcademicYear field, it will require some conversions to actually use the AcademicYear class as the key. I have refactored the code that reads value using the key to use an inline function which implies that we are reading data for a whole academic year.